webapp/channels/src/packages/mattermost-redux/src/actions/users.ts

```diff
@@ -589,59 +589,94 @@ export function getUserByEmail(email: string) {
     });
 }
 
-// We create an array to hold the id's that we want to get a status for. We build our
-// debounced function that will get called after a set period of idle time in which
-// the array of id's will be passed to the getStatusesByIds with a cb that clears out
-// the array. Helps with performance because instead of making 75 different calls for
-// statuses, we are only making one call for 75 ids.
-// We could maybe clean it up somewhat by storing the array of ids in redux state possbily?
-let ids: string[] = [];
-const debouncedGetStatusesByIds = debounce(async (dispatch: DispatchFunc) => {
-    dispatch(getStatusesByIds([...new Set(ids)]));
-}, 20, false, () => {
-    ids = [];
-});
-export function getStatusesByIdsBatchedDebounced(id: string) {
-    ids = [...ids, id];
-    return debouncedGetStatusesByIds;
-}
-
-export function getStatusesByIds(userIds: string[]) {
-    return bindClientFunc({
-        clientFunc: Client4.getStatusesByIds,
-        onSuccess: UserTypes.RECEIVED_STATUSES,
-        params: [
-            userIds,
-        ],
-    });
-}
+export function getStatusesByIds(userIds: Array<UserProfile['id']>): ActionFuncAsync<UserStatus[]> {
+    return async (dispatch, getState) => {
+        if (!userIds || userIds.length === 0) {
+            return {data: []};
+        }
 
-export function getStatus(userId: string) {
-    return bindClientFunc({
-        clientFunc: Client4.getStatus,
-        onSuccess: UserTypes.RECEIVED_STATUS,
-        params: [
-            userId,
+        let recievedStatuses: UserStatus[];
+        try {
+            recievedStatuses = await Client4.getStatusesByIds(userIds);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
+
+        const statuses: Record<UserProfile['id'], UserStatus['status']> = {};
+        const dndEndTimes: Record<UserProfile['id'], UserStatus['dnd_end_time']> = {};
+        const isManualStatuses: Record<UserProfile['id'], UserStatus['manual']> = {};
+        const lastActivity: Record<UserProfile['id'], UserStatus['last_activity_at']> = {};
+
+        for (const recievedStatus of recievedStatuses) {
+            statuses[recievedStatus.user_id] = recievedStatus?.status ?? '';
+            dndEndTimes[recievedStatus.user_id] = recievedStatus?.dnd_end_time ?? 0;
+            isManualStatuses[recievedStatus.user_id] = recievedStatus?.manual ?? false;
+            lastActivity[recievedStatus.user_id] = recievedStatus?.last_activity_at ?? 0;
+        }
+
+        dispatch(batchActions([
```

Comment:
I'm not sure if we actually need to break up the action into 4 different actions here. If we were to use the old actions, we could have the reducer handle that more efficiently without having to make a new object here. I'm thinking about a couple points from the Redux style guide here:
- https://redux.js.org/style-guide/#put-as-much-logic-as-possible-in-reducers
- https://redux.js.org/style-guide/#model-actions-as-events-not-setters

Reply:
After chatting, we decided this isn't worth it